src/main/java/com/box/sdk/BoxFile.java

```diff
@@ -927,11 +928,32 @@ public void removeWatermark() {
      * @param inputStream the stream instance that contains the data.
      * @param fileSize the size of the file that will be uploaded.
      * @return the created file instance.
+     * @throws InterruptedException when a thread execution is interrupted.
+     * @throws IOException when reading a stream throws exception.
      */
-    public BoxFile.Info uploadLargeFile(InputStream inputStream, long fileSize) {
+    public BoxFile.Info uploadLargeFile(InputStream inputStream, long fileSize)
+        throws InterruptedException, IOException {
```

Comment:
Please wrap with BoxAPIException as Interrupted exception should be masked as uploadLargeFile masks the multiple connection in the SDK